src/api/Manager.js

```diff
@@ -273,16 +276,45 @@ const API = {
   genuineCheck: (
     transport: Transport<*>,
     { targetId, perso }: { targetId: *, perso: * }
-  ) =>
+  ): Observable<GenuineCheckEvent> =>
     createDeviceSocket(transport, {
       url: URL.format({
         pathname: `${getEnv("BASE_SOCKET_URL")}/genuine`,
         query: { targetId, perso, livecommonversion }
       })
-    }).pipe(
-      last(),
-      filter(o => o.type === "result"),
-      map(o => o.payload || "")
+    }).pipe(input =>
+      Observable.create(o => {
+        let timeout;
+        let requested;
+        input.subscribe({
```

Comment:
We need to return the subscription to support unsubscribe

Reply:
Yes I saw that on documentation. I'll fix it

Reply:
How do we use the new GitHub wip feature?